PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/components/ServerExplorerToolWindowFactory.java

```diff
@@ -244,7 +254,10 @@ private JPopupMenu createPopupMenuForNode(Node node) {
     private JMenuItem createMenuItemFromNodeAction(NodeAction nodeAction) {
         final JMenuItem menuItem = new JMenuItem(nodeAction.getName());
         menuItem.setEnabled(nodeAction.isEnabled());
-        if (nodeAction.getIconPath() != null) {
+        Icon icon = nodeAction.getListeners().get(0).getIcon();
```

Comment:
Why we need to get icon from node listeners?

Reply:
so that we can configure icon variable from a listener of an action.

Reply:
updated.